File: DecoratorPattern/Python/Coffee-Hut/beverage/condiments/soy.py
from beverage.beverage import Beverage
from beverage.condiments.condiments_decorator import CondimentsDecorator


class Soy(CondimentsDecorator):
    def __init__(self, beverage: Beverage):
        self.beverage = beverage

    def get_description(self) -> str:
        return self.beverage.get_description() + ", Soy"

    def cost(self) -> float:
        return self.beverage.cost() + 0.15

File: Decorator/Java/Coffee-Hut/src/com/laxmena/beverage/condiments/Soy.java
package com.laxmena.beverage.condiments;

import com.laxmena.beverage.Beverage;

public class Soy extends CondimentDecorator {
    public Soy(Beverage beverage) {
        this.beverage = beverage;
    }
    @Override
    public double cost() {
        return this.beverage.cost() + 0.15;
    }

    @Override
    public String getDescription() {
        return this.beverage.getDescription() + ", Soy";
    }
}

